### ISSUE TITLE
destroy subcommand fails if errors exist in configuration

### ISSUE THREAD

**Issue by rbowlby** [hashicorp/terraform#2675]
I see no reason why a destroy action is coupled to the current terraform configuration that has yet to be applied.

If I introduce a configuration error into an underlying module then attempt to do a terraform destroy I get errors related to the configuration. Who cares, I'm destroying..not applying. ¯\_(ツ)_/¯

**Comment by catsby** [comment]
Hey @rbowlby – sorry for any frustration here, but I admit I'm not sure I follow. Can you illustrate with a sample configuration what you're hitting, and what you expect the behavior to be? 

Thanks!

**Comment by phinze** [comment]
Yeah I think this makes sense... basically the semantics of `terraform destroy` should change such that everything known in the state is destroyed, even if the configuration has been corrupted in some way.

The thing that makes this difficult is that currently we lean on the configuration to determine the dependency graph, which tells us what order things can be destroyed in.

So in order to accomplish this "destroy works on state without config", we'd need to talk about encoding dependency information into the stored state, which would be a big task.

Going to mark this as a core enhancement and tag it "thinking" so we can mull over the implications of doing this - thanks for the report!

/cc @mitchellh

**Comment by LeftyBC** [comment]
:+1: 

Agreed - I currently (for my proof-of-concept setup) have a script called daily-destroy.sh that is run via cron after I've gone home.  This script runs terraform destroy -force on my POC just in case I forgot to run it manually.  The idea is to keep AWS costs down by not running machines when nobody's actively developing on them.

If the terraform config is broken in any way (such as if I wasn't finished migrating something before I went home), the env grinds away on AWS all night because the scripted destroy fails.

That's my use case - it's temporary, but at least it keeps my bosses happier when they get the bill :)
